hledger-lib/Hledger/Reports/ReportOptions.hs

```diff
@@ -117,8 +117,11 @@ data ReportOpts = ReportOpts {
     -- for account transactions reports (aregister)
     ,txn_dates_      :: Bool
     -- for balance reports (bal, bs, cf, is)
-    ,balancecalc_    :: BalanceCalculation
-    ,balanceaccum_   :: BalanceAccumulation
+    ,balancecalc_    :: BalanceCalculation  -- ^ What to calculate in balance report cells
+    ,balanceaccum_   :: BalanceAccumulation -- ^ How to accumulate balance report values over time
+    ,budgetpat_      :: T.Text  -- ^ If non-empty, a case-insensitive description substring
```

Comment:
Do you think it would make more sense for this to be `Maybe T.Text`? It would keep it a bit cleaner conceptually to maintain the distinction between ‘no pattern provided’ and ‘null pattern provided’.